### ISSUE TITLE
[BUG] Error while compiling application with ConnectKitProvider in React 18.2

### ISSUE THREAD

**Issue by 70nyIT** [family/connectkit#95]
**Describe the bug**

If I include the `<ConnectKitProvider>` element, I have a white screen showing and different errors, both in the browser console and Webstorm console.

**To reproduce**

I'm using version `1.1.2` (also with `1.1.1` happens) and `wagmi@0.10.15`.

My `index.ts` is so structured

```
<React.StrictMode>
    <Provider store={store}>
      <WagmiConfig client={client}>
        <ConnectKitProvider theme="auto">
          <StyledEngineProvider injectFirst>
            <ThemeProvider theme={theme}>
              <CssBaseline />
              <App />
            </ThemeProvider>
          </StyledEngineProvider>
        </ConnectKitProvider>
      </WagmiConfig>
    </Provider>
  </React.StrictMode>
```

If I remove `<ConnectKitProvider>` it compiles with no problems, once I include it, it shows me a white page.

`<App>` is just a simple div with "Hello world" test inside.

The `client` object passed to `wagmi` is 

```typescript
const { provider, chains } = configureChains(
  [polygonMumbai],
  [
    publicProvider()
  ],
);
const client = createClient({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector({
      chains: chains
    }),
  ],
  provider
});
```

In the browser console I get the following error;
```
Uncaught TypeError: Cannot read properties of undefined (reading '0')
    at Sa (public.js:28:1)
    at renderWithHooks (react-dom.development.js:16305:1)
    at mountIndeterminateComponent (react-dom.development.js:20074:1)
    at beginWork (react-dom.development.js:21587:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)
    at beginWork$1 (react-dom.development.js:27451:1)
    at performUnitOfWork (react-dom.development.js:26557:1)
    at workLoopSync (react-dom.development.js:26466:1)
```

While in Webstorm console I get the following error
```
WARNING in ./node_modules/json-rpc-engine/node_modules/eth-rpc-errors/dist/utils.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/.../node_modules/json-rpc-engine/node_modules/eth-rpc-errors/src/utils.ts' file: Error: ENOENT: no such file or directory, open '/.../node_modules/json-rpc-engine/node_modules/eth-rpc-errors/src/utils.ts'

WARNING in ./node_modules/safe-json-utils/dist/cjs/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/.../node_modules/safe-json-utils/src/index.ts' file: Error: ENOENT: no such file or directory, open '/.../node_modules/safe-json-utils/src/index.ts'
```

**Expected behavior**

To render the page correctly, instead of a white screen.

**Screenshots**

![image](https://user-images.githubusercontent.com/12898752/215050990-5a7e6f8c-dcd4-454f-9e35-40d6fd044409.png)

**Environment details**

React 18.2 (with typescript)
Wagmi 0.10.15
ConnectKit 1.1.2 (also with 1.1.1 same problem)

`tsconfig.json` as follows

```json
{
  "compilerOptions": {
    "target": "es6",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": [
    "src"
  ]
}
```

**Comment by shubhamsaxena-devslane** [comment]
Yeah, I was having this same problem. Instead of using the custom client, when I used `getDefaultClient` the problem was solved. But, anyways, I also want to use custom client, but facing this problem.

**Comment by 70nyIT** [comment]
Can confirm what written by @shubhamsaxena-devslane (Thank you by the way!!)

The following version of the code to create `client` makes the bug disappear

```
const { provider, chains } = configureChains(
  [polygonMumbai],
  [
    publicProvider()
  ],
);
const client = createClient(
  getDefaultClient({
    chains: chains,
    provider: provider,
    autoConnect: true,
    appName: "teamArchive"
  })
);
```

But would be great to be able to inject a custom client, rather only being able to get the default one.

**Comment by Cue972** [comment]
I have exactly the same problem.

getDefaultClient works, but not custom client

**Comment by lochie** [comment]
Not sure why this is occurring from first glance.
Could you set up a codesandbox example so we can help debug?

**Comment by shubhamsaxena-devslane** [comment]
@lochie Sure, here it is.
https://codesandbox.io/p/sandbox/connectkit-vite-typescript-custom-client-hv0xtn
Please, have a look at console after opening the preview in new tab. Thanks.

**Comment by lochie** [comment]
Sorry for the delay on this, but this should be resolved in ConnectKit version `1.2.1` with wagmi `0.11.x`.
If you're still seeing this issue please let us know.
